posts/2023/03/16/kubernetes-meets-event-driven-ansible.md

```diff
@@ -0,0 +1,203 @@
+
+After confirming that the Kubernetes source plugin is capturing events successfully, let’s demonstrate how one could make use of these events within an Ansible Playbook. Create a new file called k8s-eda-demo-playbook.yaml with the following content.
+
+```
+- hosts: localhost
+  connection: local
+  tasks:
+    - ansible.builtin.debug:
+        msg: "ConfigMap in namespace '{{ event.resource.metadata.namespace }}' with name '{{ event.resource.metadata.name }} ‘{{ event.type | capitalize }}’ with the message ‘{{ event.resource.data.message}}'"
+
+```
+
+This playbook demonstrates how to obtain properties that are included on the captured event. The type property will display “Added” as the playbook will only execute when ConfigMaps that have been created. The ConfigMap object itself can be accessed by referencing the resource property on the event. The standard Kubernetes manifest for a ConfigMap can then be traversed, such as the _namespace, name_ as well as specific *data values*. 
+
+Update the contents of the rulebook in the k8s-eda-demo.yaml file to invoke the newly created playbook instead of simply printing out the contents by using the run_playbook action as shown below:
+
+```
+- name: Listen for newly added ConfigMap resources
+  hosts: all
+  sources:
+    - sabre1041.eda.k8s:
+        api_version: v1
+        kind: ConfigMap
+        namespace: default
+  rules:
+    - name: Execute Playbook
+      condition: event.type == "ADDED"
+      action:
+        run_playbook:
+          name: k8s-eda-demo-playbook.yaml   
+```
+
+Once again, execute the k8s-eda-demo-playbook.yaml rulebook to begin listing for ConfigMap’s added to the _default_ namespace.
+
+```
+ansible-rulebook -i inventory --rulebook k8s-eda-demo.yaml --verbose 
+```
+
+Delete and recreate the ConfigMap to trigger the playbook.
+
+```
+kubectl delete configmap -n default eda-example
+
+kubectl create configmap -n default eda-example --from-literal=message=”Kubernetes Meets Event-Driven Ansible”
+```
+
+Observe that the playbook has been triggered and produces output similar to the following:
+
+```
+TASK [ansible.builtin.debug] ***************************************************
+ok: [localhost] => {
+    "msg": "ConfigMap in namespace 'default' with name 'eda-example ‘Added’ with the message ‘Kubernetes Meets Event-Driven Ansible'"
+}
+```
+
+While this example only prints out a simple message related to the content of the event received, it provides a demonstration of how to make use of the capabilities enabled by the Kubernetes integration. By adding Kuberentes as an event source and into the Event-Driven Ansible ecosystem, it becomes an essential integration to help support organizations that are leveraging Kubernetes to maintain crucial components of their business and to trigger automation as desired.
+
+__Topics:__
+
+[Event-Driven Ansible](https://www.ansible.com/blog/topic/event-driven-ansible)
```

Comment:
We should remove this or change the link to the "event-driven-ansible" tag in the Nikola format:
`https://ansible.com/blog/categories/event-driven-ansible/`

Note: that's assuming we keep the ansible.com/blog path and don't move to a subdomain (not sure it's possible to use Nikola for the main page and a subdomain at the same time)